packages/aws-cdk-lib/aws-stepfunctions/README.md

```diff
@@ -975,6 +975,77 @@ new sfn.StateMachine(this, 'MyStateMachine', {
 });
 ```
 
+## Encryption 
+You can encrypt your data using a customer-managed key for AWS Step Functions state machines and activities. You can configure a symmetric AWS KMS key and data key reuse period when creating or updating a State Machine, and when creating an Activity. The execution history and state machine definition will be encrypted with the key applied to the State Machine. Activity inputs will be encrypted with the key applied to the Activity.
+
+### Encrypting state machines 
+You can provide a symmetric KMS key to encrypt the state machine definition and execution history:
+```
+const kmsKey = new kms.Key(this, 'Key');
+const stateMachine = new sfn.StateMachine(this, 'StateMachineWithCMKEncryptionConfiguration', {
+  stateMachineName: 'StateMachineWithCMKEncryptionConfiguration',
+  definitionBody: sfn.DefinitionBody.fromChainable(sfn.Chain.start(new sfn.Pass(this, 'Pass'))),
+  stateMachineType: sfn.StateMachineType.STANDARD,
+  kmsKey: kmsKey,
+  kmsDataKeyReusePeriodSeconds: cdk.Duration.seconds(60),
+});
+```
+
+### Creating a StateMachine with CWL Encryption using a Customer Managed Key,
+If a state machine is encrypted with a customer managed key and has logging enabled, its decrypted execution history will be stored in CloudWatch Logs. To encrypt the logs with your own KMS key you must encrypt the `LogGroup` used by the state machine with a KMS key.
+```
+const stateMachineKmsKey = new kms.Key(this, 'StateMachine Key');
+const logGroupKey = new kms.Key(this, 'LogGroup Key');
+
+// To enable encrypted CloudWatch log integration for a state machine, you must add the following to your log group key policy
```

Comment:
This comment is somewhat misleading. The policy is just to set up an encryptied CloudWatch Logs log group. Nothing to do with the state machine really

Reply:
Perhaps we can omit the comment entirely or trim it down:

```suggestion
// Required KMS key policy to enrypt the CloudWatch log group
```